src/Microsoft.AspNetCore.OData/EnableQueryAttribute.cs

```diff
@@ -228,46 +228,44 @@ public override void OnActionExecuted(ActionExecutedContext actionExecutedContex
             {
                 // actionExecutedContext.Result might also indicate a status code that has not yet
                 // been applied to the result; make sure it's also successful.
-                StatusCodeResult statusCodeResult = actionExecutedContext.Result as StatusCodeResult;
-                if (statusCodeResult == null || IsSuccessStatusCode(statusCodeResult.StatusCode))
+                ObjectResult responseContent = actionExecutedContext.Result as ObjectResult;
+
+                if (responseContent != null && (responseContent.StatusCode == null || IsSuccessStatusCode(responseContent.StatusCode.Value)))
                 {
-                    ObjectResult responseContent = actionExecutedContext.Result as ObjectResult;
-                    if (responseContent != null)
+
+                    //throw Error.Argument("actionExecutedContext", SRResources.QueryingRequiresObjectContent,
+                    //    actionExecutedContext.Result.GetType().FullName);
```

Comment:
Maybe you could remove this commented code.

It seems hat it as meant to throw an exception if the result was not an ObjectResult. I wonder why it was commented out. But I think it should be removed if it's not used.